services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/S3JavaMultipartTransferProgressListenerTest.java

```diff
@@ -0,0 +1,199 @@
+ */
+
+package software.amazon.awssdk.transfer.s3.internal;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletionException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.NoSuchBucketException;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import software.amazon.awssdk.testutils.RandomTempFile;
+import software.amazon.awssdk.transfer.s3.CaptureTransferListener;
+import software.amazon.awssdk.transfer.s3.S3TransferManager;
+import software.amazon.awssdk.transfer.s3.model.FileUpload;
+import software.amazon.awssdk.transfer.s3.progress.LoggingTransferListener;
+import software.amazon.awssdk.transfer.s3.progress.TransferListener;
+
+@WireMockTest
+public class S3JavaMultipartTransferProgressListenerTest {
+
+    public static final String ERROR_CODE = "NoSuchBucket";
+    public static final String ERROR_MESSAGE = "We encountered an internal error. Please try again.";
+    public static final String ERROR_BODY = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                                            + "<Error>\n"
+                                            + "  <Code>" + ERROR_CODE + "</Code>\n"
+                                            + "  <Message>" + ERROR_MESSAGE + "</Message>\n"
+                                            + "</Error>";
+    private static final String EXAMPLE_BUCKET = "Example-Bucket";
+    private static final String TEST_KEY = "16mib_file.dat";
+    private static final int OBJ_SIZE = 16 * 1024 * 1024;
+    private static S3AsyncClient mpuClient;
+    private RandomTempFile testFile;
+
+    @BeforeAll
+    public static void init(WireMockRuntimeInfo wm) {
+        mpuClient = S3AsyncClient.builder()
```

Comment:
Can we add parameterized tests for TM with S3 client that doesn't have multipart enabled?